testsuite/henderson_hasselbalch_tests.py

```diff
@@ -16,87 +16,72 @@
-    np.savetxt(f, np.asarray(HH_Donnan_dict["charges_dict"]["peptide_2"]).reshape(1,-1), delimiter=",")
-"""
-
-ref_data_HH_Donnan = np.loadtxt(f"{data_path}/HH_Donnan.csv", delimiter=",")
-np.testing.assert_allclose(HH_Donnan_dict["charges_dict"]["peptide_1"], ref_data_HH_Donnan[0,:])
-np.testing.assert_allclose(HH_Donnan_dict["charges_dict"]["peptide_2"], ref_data_HH_Donnan[1,:])
-
-print("*** Test passed ***\n")
-
-
-print("*** Check that HH and HH_Don are consistent ***")
-
-Z_HH_1 = pmb.calculate_HH(molecule_name = "peptide_1", 
-                        pH_list = HH_Donnan_dict["pH_system_list"])
-Z_HH_2 = pmb.calculate_HH(molecule_name = "peptide_2", 
-                        pH_list = HH_Donnan_dict["pH_system_list"])
-
-np.testing.assert_allclose(Z_HH_1, HH_Donnan_dict["charges_dict"]["peptide_1"])
-np.testing.assert_allclose(Z_HH_2, HH_Donnan_dict["charges_dict"]["peptide_2"])
-
-
-print("*** Test passed***")
+class Test(ut.TestCase):
+    data_root = pathlib.Path(__file__).parent.resolve() / "henderson_hasselbalch_tests_data"
+
+    def test(self):
+        pmb = pyMBE.pymbe_library(seed=42)
+        print("*** Running HH tests ***\n")
+
+        # Peptide parameters
+        sequence1 = 5 * "D" + 8 * "H"
+        sequence2 = 3 * "E" + 7 * "R"
+        pep1_concentration = 1e-2 * pmb.units.mol/pmb.units.L
+        pep2_concentration = 1e-2 * pmb.units.mol/pmb.units.L
+        c_salt=5e-3 * pmb.units.mol/ pmb.units.L
+        model = '1beadAA'
+
+        # Load pKa-values
+        path_to_pka=pmb.get_resource("parameters/pka_sets/Nozaki1967.json")
+        pmb.load_pka_set(path_to_pka)
+
+        # Define the peptides in the pyMBE data frame
+        pmb.define_peptide(name = "peptide_1",
+                sequence = sequence1,
+                model = model)
+
+        pmb.define_peptide(name = "peptide_2",
+                sequence = sequence2,
+                model = model)
+
+        with self.subTest(msg="Check Henderson-Hasselbalch equation"):
+            pH_range = np.linspace(2, 12, num=200)
+            Z_HH_1 = pmb.calculate_HH(molecule_name = "peptide_1",
+                                      pH_list = pH_range)
+            Z_HH_2 = pmb.calculate_HH(molecule_name = "peptide_2",
+                                      pH_list = pH_range)
+
+            data_path = pmb.get_resource(path=self.data_root)
+            ref_data_HH = np.loadtxt(f"{data_path}/HH.csv", delimiter=",")
+            np.testing.assert_allclose(Z_HH_1, ref_data_HH[0,:])
+            np.testing.assert_allclose(Z_HH_2, ref_data_HH[1,:])
+
+        with self.subTest(msg="Check Henderson-Hasselbalch equation + Donnan"):
+            HH_Donnan_dict = pmb.calculate_HH_Donnan(
+                    c_macro = {"peptide_1": pep1_concentration,
+                               "peptide_2": pep2_concentration},
+                    c_salt = c_salt,
+                    pH_list = pH_range)
```

Comment:
I took the liberty of replacing print statements by a corresponding `self.subTest()` context. This way, the message is only printed to the terminal when the assertion fails. Also, if the assertion fails, the test continues, so you can see how many equations are wrong. You no longer need to run the test multiple times with commented out sections to determine how many features are affected by a regression.

Note that `self.subTest()` is only useful when you cannot split assertions across multiple `def test_*()` functions, for example because all assertions rely on common data that is expensive to compute. If the data is cheap to generate, then it's preferable to use `def setUp()` and `def tearDown()` to set up the data before each test and clean it up at the end of each test. `tearDown()` will run even if the test failed.